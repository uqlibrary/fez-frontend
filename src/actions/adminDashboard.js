import * as actions from './actionTypes';
import { get, post, put, destroy } from 'repositories/generic';
import {
    ADMIN_DASHBOARD_CONFIG_API,
    ADMIN_DASHBOARD_TODAY_API,
    ADMIN_DASHBOARD_QUICKLINKS_API,
    ADMIN_DASHBOARD_SYSTEM_ALERTS_API,
    ADMIN_DASHBOARD_DISPLAY_REPORT_API,
    ADMIN_DASHBOARD_EXPORT_REPORT_API,
} from 'repositories/routes';

import { promptForDownload } from './exportPublicationsDataTransformers';

/**
 * Fetches the config data for admin dashboard
 * @returns {function(*)}
 */
export function loadAdminDashboardConfig() {
    return dispatch => {
        dispatch({
            type: actions.ADMIN_DASHBOARD_CONFIG_LOADING,
        });
        return get(ADMIN_DASHBOARD_CONFIG_API())
            .then(response => {
                dispatch({
                    type: actions.ADMIN_DASHBOARD_CONFIG_SUCCESS,
                    payload: response,
                });
                return Promise.resolve(response);
            })
            .catch(error => {
                console.error(error.message);
                dispatch({
                    type: actions.ADMIN_DASHBOARD_CONFIG_FAILED,
                    payload: error.message,
                });
                return Promise.reject(error);
            });
    };
}

/**
 * Fetches the metrics for the Today tab
 * @returns {function(*)}
 */
export function loadAdminDashboardToday() {
    return dispatch => {
        dispatch({
            type: actions.ADMIN_DASHBOARD_TODAY_LOADING,
        });
        return get(ADMIN_DASHBOARD_TODAY_API())
            .then(response => {
                dispatch({
                    type: actions.ADMIN_DASHBOARD_TODAY_SUCCESS,
                    payload: response,
                });
            })
            .catch(error => {
                dispatch({
                    type: actions.ADMIN_DASHBOARD_TODAY_FAILED,
                    payload: error.message,
                });
            });
    };
}

/**
 * Fetches the quicklinks
 * @returns {function(*)}
 */
export function loadAdminDashboardQuickLinks() {
    return dispatch => {
        dispatch({
            type: actions.ADMIN_DASHBOARD_QUICKLINKS_LOADING,
        });
        return get(ADMIN_DASHBOARD_QUICKLINKS_API())
            .then(response => {
                dispatch({
                    type: actions.ADMIN_DASHBOARD_QUICKLINKS_SUCCESS,
                    payload: response,
                });
            })
            .catch(error => {
                dispatch({
                    type: actions.ADMIN_DASHBOARD_QUICKLINKS_FAILED,
                    payload: error.message,
                });
            });
    };
}

/**
 * Handles admin operations on quick links (edit, delete)
 * @returns {function(*)}
 */
export function adminDashboardQuickLink(request, action) {
    // eslint-disable-next-line no-nested-ternary
    const verb = action === 'DELETE' ? destroy : action === 'EDIT' || action === 'REORDER' ? put : post;

    return dispatch => {
        dispatch({ type: actions.ADMIN_DASHBOARD_QUICKLINKS_UPDATING });

        return verb(ADMIN_DASHBOARD_QUICKLINKS_API(), request)
            .then(response => {
                dispatch({
                    type: actions.ADMIN_DASHBOARD_QUICKLINKS_UPDATE_SUCCESS,
                    payload: response,
                });
                return Promise.resolve(response);
            })
            .catch(error => {
                dispatch({
                    type: actions.ADMIN_DASHBOARD_QUICKLINKS_UPDATE_FAILED,
                    payload: error.message,
                });
                return Promise.reject(error);
            });
    };
}

export function adminDashboardQuickLinkUpdateClear() {
    return dispatch => {
        dispatch({
            type: actions.ADMIN_DASHBOARD_QUICKLINKS_UPDATE_CLEAR,
        });
    };
}

/**
 * Fetches the System Alerts
 * @returns {function(*)}
 */
export function loadAdminDashboardSystemAlerts() {
    return dispatch => {
        dispatch({
            type: actions.ADMIN_DASHBOARD_SYSTEM_ALERTS_LOADING,
        });
        return get(ADMIN_DASHBOARD_SYSTEM_ALERTS_API())
            .then(response => {
                dispatch({
                    type: actions.ADMIN_DASHBOARD_SYSTEM_ALERTS_SUCCESS,
                    payload: response,
                });
            })
            .catch(error => {
                dispatch({
                    type: actions.ADMIN_DASHBOARD_SYSTEM_ALERTS_FAILED,
                    payload: error.message,
                });
            });
    };
}

/**
 * Handles admin operations on system alerts (assign, resolve)
 * @returns {function(*)}
 */
export function adminDashboardSystemAlerts(request) {
    return dispatch => {
        dispatch({ type: actions.ADMIN_DASHBOARD_SYSTEM_ALERT_UPDATING });
        return put(ADMIN_DASHBOARD_SYSTEM_ALERTS_API(), request)
            .then(response => {
                dispatch({
                    type: actions.ADMIN_DASHBOARD_SYSTEM_ALERT_UPDATE_SUCCESS,
                    payload: response,
                });

                return Promise.resolve(response);
            })
            .catch(error => {
                dispatch({
                    type: actions.ADMIN_DASHBOARD_SYSTEM_ALERT_UPDATE_FAILED,
                    payload: error.message,
                });
                return Promise.reject(error);
            });
    };
}

export function adminDashboardSystemAlertsUpdateClear() {
    return dispatch => {
        dispatch({
            type: actions.ADMIN_DASHBOARD_SYSTEM_ALERTS_UPDATE_CLEAR,
        });
    };
}

/**
 * Fetches export (legacy) report data as a file attachment
 * or queues the request at the server.
 * @returns {function(*)}
 */
export function loadAdminDashboardExportReport(request, options) {
    return dispatch => {
        const exportConfig = {
            format: options.export_to,
        };

        dispatch({
            type: actions.ADMIN_DASHBOARD_EXPORT_REPORT_LOADING,
        });

        return get(ADMIN_DASHBOARD_EXPORT_REPORT_API(request))
            .then(response => {
                // only try to export to csv if we get a text response
                if (typeof response === 'string') {
                    const blobResponse = new Blob([response], { type: 'text/csv' });
                    promptForDownload(exportConfig.format, blobResponse);
                }

                dispatch({
                    type: actions.ADMIN_DASHBOARD_EXPORT_REPORT_SUCCESS,
                    payload: exportConfig,
                });

                return Promise.resolve(response);
            })
            .catch(error => {
                console.error(error);
                dispatch({
                    type: actions.ADMIN_DASHBOARD_EXPORT_REPORT_FAILED,
                    payload: {
                        ...exportConfig,
                        errorMessage: error.message,
                    },
                });
            });
    };
}

/**
 * Fetches display report data
 * @returns {function(*)}
 */
export function loadAdminDashboardDisplayReport(request) {
    return dispatch => {
        dispatch({
            type: actions.ADMIN_DASHBOARD_DISPLAY_REPORT_LOADING,
            params: { ...request },
        });
        return get(ADMIN_DASHBOARD_DISPLAY_REPORT_API(request))
            .then(response => {
                dispatch({
                    type: actions.ADMIN_DASHBOARD_DISPLAY_REPORT_SUCCESS,
                    payload: response,
                });
            })
            .catch(error => {
                dispatch({
                    type: actions.ADMIN_DASHBOARD_DISPLAY_REPORT_FAILED,
                    payload: {
                        errorMessage: error.message,
                    },
                });
            });
    };
}

export function clearAdminDashboardDisplayReport() {
    return dispatch => {
        dispatch({
            type: actions.ADMIN_DASHBOARD_DISPLAY_REPORT_CLEAR,
        });
    };
}

export function clearAdminDashboardExportReport() {
    return dispatch => {
        dispatch({
            type: actions.ADMIN_DASHBOARD_EXPORT_REPORT_CLEAR,
        });
    };
}
