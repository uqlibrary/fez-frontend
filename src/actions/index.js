export * from './academic';
export * from './account';
export * from './app';
export * from './authors';
export * from './batchImport';
export * from './bulkUpdates';
export * from './claimPublications';
export * from './controlledVocabularies';
export * from './deleteRecord';
export * from './doi';
export * from './exportPublications';
export * from './favouriteSearch';
export * from './fixRecord';
export * from './incompleteRecords';
export * from './issnLinks';
export * from './journals';
export * from './manageAuthors';
export * from './manageUsers';
export * from './myEditorialAppointments';
export * from './orcidSync';
export * from './organisationalUnits';
export * from './publications';
export * from './records';
export * from './search';
export * from './thirdPartyLookupTool';
export * from './viewRecord';
export * from './viewCommunitiesCollections';
export * from './adminScaleSignificance';
export * from './viewControlledVocab';
