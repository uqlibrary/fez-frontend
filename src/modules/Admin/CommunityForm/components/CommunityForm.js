import React from 'react';
import PropTypes from 'prop-types';
import { propTypes } from 'redux-form/immutable';
import { Field } from 'redux-form/immutable';

import { Alert } from 'modules/SharedComponents/Toolbox/Alert';
import { NavigationDialogBox } from 'modules/SharedComponents/Toolbox/NavigationPrompt';
import { TextField } from 'modules/SharedComponents/Toolbox/TextField';
import { RichEditorField } from 'modules/SharedComponents/RichEditor';
import { StandardPage } from 'modules/SharedComponents/Toolbox/StandardPage';
import { StandardCard } from 'modules/SharedComponents/Toolbox/StandardCard';
import { NewListEditorField, KeywordsForm } from 'modules/SharedComponents/Toolbox/ListEditor';
import { ConfirmDiscardFormChanges } from 'modules/SharedComponents/ConfirmDiscardFormChanges';
import { validation } from 'config';
import { default as formLocale } from 'locale/publicationForm';
import Button from '@mui/material/Button';
import Grid from '@mui/material/Grid';
import Typography from '@mui/material/Typography';
import { pathConfig } from 'config/pathConfig';

export const CommunityForm = ({ disableSubmit, newRecord, ...props }) => {
    const cancelSubmit = () => {
        window.location.assign(pathConfig.index);
    };

    const afterSubmit = () => {
        window.location.assign(pathConfig.index);
    };

    const reloadForm = () => {
        window.location.reload();
    };

    const txt = formLocale.addACommunity;
    if (props.submitSucceeded && newRecord) {
        return (
            <StandardPage title={txt.title}>
                <Grid container spacing={3}>
                    <Grid item xs={12}>
                        <StandardCard title={txt.afterSubmitTitle}>
                            <Typography>{txt.afterSubmitText}</Typography>
                        </StandardCard>
                    </Grid>
                </Grid>
                <Grid container spacing={2}>
                    <Grid item xs />
                    <Grid item>
                        <Button variant="contained" fullWidth onClick={afterSubmit}>
                            {txt.afterSubmitButton}
                        </Button>
                    </Grid>
                    <Grid item>
                        <Button variant="contained" color="primary" fullWidth onClick={reloadForm}>
                            {txt.AddAnotherButton}
                        </Button>
                    </Grid>
                </Grid>
            </StandardPage>
        );
    }
    // customise error for thesis submission
    const alertProps = validation.getErrorAlertProps({
        ...props,
        alertLocale: {
            validationAlert: { ...formLocale.validationAlert },
            progressAlert: { ...formLocale.progressAlert },
            successAlert: { ...formLocale.successAlert },
            errorAlert: {
                ...formLocale.errorAlert,
                message: formLocale.addACommunity.addFailedMessage,
            },
        },
    });
    return (
        <StandardPage title={txt.title}>
            <ConfirmDiscardFormChanges dirty={props.dirty} submitSucceeded={props.submitSucceeded}>
                <form>
                    <NavigationDialogBox
                        when={props.dirty && !props.submitSucceeded}
                        txt={txt.cancelWorkflowConfirmation}
                    />
                    <Grid container spacing={3}>
                        <Grid item xs={12}>
                            <Grid item xs={12}>
                                <StandardCard title={txt.details.title} help={txt.details.help}>
                                    <Grid container spacing={3}>
                                        <Grid item xs={12}>
                                            <Field
                                                component={TextField}
                                                textFieldId="rek-title"
                                                disabled={props.submitting}
                                                autoFocus
                                                name="rek_title"
                                                type="text"
                                                fullWidth
                                                {...txt.formLabels.title}
                                                required
                                                validate={[validation.required]}
                                            />
                                        </Grid>

                                        <Grid item xs={12}>
                                            <Field
                                                component={TextField}
                                                textFieldId="rek-description"
                                                disabled={props.submitting}
                                                name="rek_description"
                                                fullWidth
                                                multiline
                                                rows={5}
                                                {...txt.formLabels.description}
                                                validate={[validation.required]}
                                                required
                                            />
                                        </Grid>

                                        <Grid item xs={12}>
                                            <Typography>{txt.formLabels.keywords.description}</Typography>
                                            <Field
                                                component={NewListEditorField}
                                                name="fez_record_search_key_keywords"
                                                maxCount={10}
                                                // isValid={validation.isValidKeyword(111)}
                                                searchKey={{ value: 'rek_keywords', order: 'rek_keywords_order' }}
                                                listEditorId="rek-keywords"
                                                locale={txt.formLabels.keywords.field}
                                                disabled={props.submitting}
                                                ListEditorForm={KeywordsForm}
                                            />
                                        </Grid>

                                        <Grid item xs={12}>
                                            <Typography>{txt.formLabels.internalNotes.label}</Typography>
                                            <Field
                                                component={RichEditorField}
                                                richEditorId="internalNotes"
                                                disabled={props.submitting}
                                                name="internalNotes"
                                                fullWidth
                                                multiline
                                                rows={5}
                                                {...txt.formLabels.internalNotes}
                                            />
                                        </Grid>
                                    </Grid>
                                </StandardCard>
                            </Grid>
                            {alertProps && (
                                <Grid item xs={12}>
                                    <Alert {...alertProps} />
                                </Grid>
                            )}
                        </Grid>

                        <Grid container spacing={2}>
                            <Grid item xs={false} sm />
                            <Grid item xs={12} sm="auto">
                                <Button
                                    data-analyticsid="cancel-community"
                                    data-testid="cancel-community"
                                    variant="contained"
                                    fullWidth
                                    disabled={props.submitting}
                                    onClick={cancelSubmit}
                                >
                                    {txt.cancel}
                                </Button>
                            </Grid>

                            <Grid item xs={12} sm="auto">
                                <Button
                                    data-analyticsid="submit-community"
                                    data-testid="submit-community"
                                    variant="contained"
                                    color="primary"
                                    fullWidth
                                    onClick={props.handleSubmit}
                                    disabled={props.submitting || disableSubmit}
                                >
                                    {txt.submit}
                                </Button>
                            </Grid>
                        </Grid>
                    </Grid>
                </form>
            </ConfirmDiscardFormChanges>
        </StandardPage>
    );
};
CommunityForm.propTypes = {
    ...propTypes, // all redux-form props
    author: PropTypes.object,
    account: PropTypes.bool,
    disableSubmit: PropTypes.bool,
    fileAccessId: PropTypes.number,
    actions: PropTypes.object,
    isSessionValid: PropTypes.bool,
    formValues: PropTypes.object,
    formErrors: PropTypes.object,

    newCommunitySaving: PropTypes.bool,
    newCommunityError: PropTypes.bool,
    newRecord: PropTypes.object,
};
export default CommunityForm;
