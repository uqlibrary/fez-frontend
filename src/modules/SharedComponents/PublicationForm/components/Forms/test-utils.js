import { useForm } from 'react-hook-form';
import { render, WithReduxStore } from 'test-utils';
import React from 'react';
import { formValues } from '../PublicationForm';

export const setupWrapper = (Form, props) => {
    const Wrapper = () => {
        const {
            control,
            formState: { isSubmitting },
        } = useForm();

        // eslint-disable-next-line react/prop-types
        return <Form {...{ control, isSubmitting, ...props, formValues: formValues(props?.formValues || {}) }} />;
    };

    return render(
        <WithReduxStore>
            <Wrapper />
        </WithReduxStore>,
    );
};
