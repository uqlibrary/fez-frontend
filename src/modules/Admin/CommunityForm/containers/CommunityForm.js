import { connect } from 'react-redux';
import { reduxForm, getFormValues, SubmissionError, getFormSyncErrors } from 'redux-form/immutable';
import Immutable from 'immutable';
import CommunityForm from '../components/CommunityForm';
import { createCommunity, checkSession, clearSessionExpiredFlag } from 'actions';
import { bindActionCreators } from 'redux';
import { getNotesSectionSearchKeys } from 'actions/transformers';

const FORM_NAME = 'Community';

const onSubmit = (values, dispatch, props) => {
    const dataJS = { ...values.toJS() };
    const data = { ...dataJS, ...getNotesSectionSearchKeys(dataJS) };

    delete data.internalNotes; // transformed above to fez_internal_notes: {ain_detail}

    const currentAuthor = props.author || null;
    // eslint-disable-next-line camelcase
    return dispatch(createCommunity(data, currentAuthor?.aut_id || null)).catch(error => {
        throw new SubmissionError({ _error: error });
    });
};

const CommunityContainer = reduxForm({
    form: FORM_NAME,
    onSubmit,
})(CommunityForm);

const mapStateToProps = state => {
    const formErrors = getFormSyncErrors(FORM_NAME)(state) || Immutable.Map({});
    return {
        formValues: getFormValues(FORM_NAME)(state) || Immutable.Map({}),
        formErrors: formErrors,
        disableSubmit: formErrors && !(formErrors instanceof Immutable.Map),
        author: (state && state.get('accountReducer') && state.get('accountReducer').author) || null,
        newRecord:
            (state && state.get('createCommunityReducer') && state.get('createCommunityReducer').newRecord) || null,
    };
};

const mapDispatchToProps = dispatch => ({
    actions: bindActionCreators({ checkSession, clearSessionExpiredFlag }, dispatch),
});

export default connect(mapStateToProps, mapDispatchToProps)(CommunityContainer);
