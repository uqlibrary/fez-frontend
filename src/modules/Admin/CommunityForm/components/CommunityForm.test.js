import CommunityForm from './CommunityForm';
import Immutable from 'immutable';
import React from 'react';
import { render, WithReduxStore, WithRouter, fireEvent, userEvent, waitFor, preview } from 'test-utils';
import * as actions from 'actions';

jest.mock('actions', () => ({
    ...jest.requireActual('actions'), // Retain the actual implementations of other functions
    createCommunity: jest.fn(),
}));

function setup(testProps = {}, state = {}, rerender = render) {
    return rerender(
        <WithReduxStore initialState={Immutable.Map(state)}>
            <WithRouter>
                <CommunityForm {...testProps} />
            </WithRouter>
        </WithReduxStore>,
    );
}
async function inputText(getByTestId, settings) {
    for (const [testId, value] of settings) {
        const input = getByTestId(testId);
        await userEvent.click(input);
        await userEvent.type(input, value);
        await userEvent.tab();
        expect(input).toHaveValue(value);
    }
}

describe('Community form', () => {
    const { location } = window;

    beforeAll(() => {
        delete window.location;
        window.location = { assign: jest.fn(), reload: jest.fn() };
    });

    afterAll(() => {
        window.location = location;
    });

    it('should render form', async () => {
        actions.createCommunity
            .mockImplementationOnce(() => {
                return () => Promise.reject(new Error('test'));
            })
            .mockImplementationOnce(() => {
                return () => Promise.resolve();
            });

        const { getByTestId, rerender } = setup();
        await inputText(getByTestId, [
            ['rek-title-input', 'test'],
            ['rek-description-input', 'test'],
        ]);
        const submitButton = getByTestId('submit-community');
        await waitFor(() => expect(submitButton).toBeEnabled());

        await userEvent.click(submitButton);
        await waitFor(() => expect(getByTestId('api-error-alert')).toBeInTheDocument());

        setup(
            {},
            {
                createCommunityReducer: {
                    newRecord: true,
                },
            },
            rerender,
        );

        await userEvent.click(submitButton);
        preview.debug();
        await waitFor(() => expect(getByTestId('after-submit-community')).toBeInTheDocument());
        await userEvent.click(getByTestId('after-submit-community'));
        expect(window.location.assign).toHaveBeenCalledWith('/');

        await userEvent.click(getByTestId('reload-community'));
        expect(window.location.reload).toHaveBeenCalled();
    });

    it('should redirect to cancel page', () => {
        const { getByTestId } = setup({});
        fireEvent.click(getByTestId('cancel-community'));
        expect(window.location.assign).toHaveBeenCalledWith('/');
    });
});
