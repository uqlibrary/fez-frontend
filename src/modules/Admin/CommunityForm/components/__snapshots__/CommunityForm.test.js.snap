// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Community form should display successfull submission screen 1`] = `
<StandardPage
  title="Add a missing community"
>
  <ConfirmDiscardFormChanges
    dirty={true}
    submitSucceeded={true}
  >
    <form>
      <NavigationDialogBox
        when={false}
      />
      <WithStyles(ForwardRef(Grid))
        container={true}
        spacing={3}
      >
        <WithStyles(ForwardRef(Grid))
          item={true}
          xs={12}
        >
          <WithStyles(ForwardRef(Grid))
            item={true}
            xs={12}
          >
            <StandardCard
              help={Object {}}
              title="Community details"
            >
              <WithStyles(ForwardRef(Grid))
                container={true}
                spacing={3}
              >
                <WithStyles(ForwardRef(Grid))
                  item={true}
                  xs={12}
                >
                  <Field
                    autoFocus={true}
                    component={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "propTypes": Object {
                          "classes": [Function],
                          "help": [Function],
                          "innerRef": [Function],
                          "textFieldId": [Function],
                        },
                        "render": [Function],
                      }
                    }
                    disabled={false}
                    fullWidth={true}
                    label="Title of community"
                    name="rek_title"
                    placeholder="Type a title for this community"
                    required={true}
                    textFieldId="rek-title"
                    type="text"
                    validate={
                      Array [
                        [Function],
                      ]
                    }
                  />
                </WithStyles(ForwardRef(Grid))>
                <WithStyles(ForwardRef(Grid))
                  item={true}
                  xs={12}
                >
                  <Field
                    component={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "propTypes": Object {
                          "classes": [Function],
                          "help": [Function],
                          "innerRef": [Function],
                          "textFieldId": [Function],
                        },
                        "render": [Function],
                      }
                    }
                    disabled={false}
                    fullWidth={true}
                    label="Community description"
                    multiline={true}
                    name="rek_description"
                    placeholder="Type a description of this community."
                    required={true}
                    rows={5}
                    textFieldId="rek-description"
                    validate={
                      Array [
                        [Function],
                      ]
                    }
                  />
                </WithStyles(ForwardRef(Grid))>
                <WithStyles(ForwardRef(Grid))
                  item={true}
                  xs={12}
                >
                  <WithStyles(ForwardRef(Typography))>
                    Add up to 10 individual keywords, or a pipe separated list, that describe the content of the thesis. (eg. one|two|three)
                  </WithStyles(ForwardRef(Typography))>
                  <Field
                    ListEditorForm={
                      Object {
                        "$$typeof": Symbol(react.memo),
                        "compare": null,
                        "type": [Function],
                      }
                    }
                    component={
                      Object {
                        "$$typeof": Symbol(react.memo),
                        "compare": null,
                        "type": [Function],
                      }
                    }
                    disabled={false}
                    listEditorId="rek-keywords"
                    locale={
                      Object {
                        "form": Object {
                          "locale": Object {
                            "addButtonLabel": "Add",
                            "id": "keywords-input",
                            "inputFieldHint": "Type keywords",
                            "inputFieldLabel": "Keywords",
                          },
                        },
                        "header": Object {
                          "locale": Object {
                            "deleteAll": "Remove all keywords",
                            "deleteAllConfirmation": Object {
                              "cancelButtonLabel": "No",
                              "confirmButtonLabel": "Yes",
                              "confirmationMessage": "Are you sure you want to delete all keywords?",
                              "confirmationTitle": "Delete all",
                            },
                            "nameColumn": "Keyword",
                            "reorderColumn": "Reorder keywords",
                          },
                        },
                        "row": Object {
                          "locale": Object {
                            "deleteHint": "Remove this keyword",
                            "deleteRecordConfirmation": Object {
                              "cancelButtonLabel": "No",
                              "confirmButtonLabel": "Yes",
                              "confirmationMessage": "Are you sure you want to delete this keyword?",
                              "confirmationTitle": "Delete keyword",
                            },
                            "moveDownHint": "Move keyword down the order",
                            "moveUpHint": "Move keyword up the order",
                          },
                        },
                      }
                    }
                    maxCount={10}
                    name="fez_record_search_key_keywords"
                    searchKey={
                      Object {
                        "order": "rek_keywords_order",
                        "value": "rek_keywords",
                      }
                    }
                  />
                </WithStyles(ForwardRef(Grid))>
                <WithStyles(ForwardRef(Grid))
                  item={true}
                  xs={12}
                >
                  <WithStyles(ForwardRef(Typography))>
                    Internal Notes (admin)
                  </WithStyles(ForwardRef(Typography))>
                  <Field
                    component={[Function]}
                    disabled={false}
                    fullWidth={true}
                    label="Internal Notes (admin)"
                    multiline={true}
                    name="internalNotes"
                    placeholder="Optionally enter internal notes"
                    richEditorId="internalNotes"
                    rows={5}
                  />
                </WithStyles(ForwardRef(Grid))>
              </WithStyles(ForwardRef(Grid))>
            </StandardCard>
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid))
            item={true}
            xs={12}
          >
            <Alert
              allowDismiss={false}
              customIcon={null}
              customType={null}
              disableAlertClick={false}
              dismissTitle="Click to dismiss this alert"
              message="New work has been saved successfully."
              showLoader={false}
              title="Success"
              type="done"
              wiggle={null}
            />
          </WithStyles(ForwardRef(Grid))>
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          container={true}
          spacing={2}
        >
          <WithStyles(ForwardRef(Grid))
            item={true}
            sm={true}
            xs={false}
          />
          <WithStyles(ForwardRef(Grid))
            item={true}
            sm="auto"
            xs={12}
          >
            <WithStyles(ForwardRef(Button))
              data-testid="cancel-community"
              disabled={false}
              fullWidth={true}
              onClick={[Function]}
              variant="contained"
            >
              Cancel and return to the homepage
            </WithStyles(ForwardRef(Button))>
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid))
            item={true}
            sm="auto"
            xs={12}
          >
            <WithStyles(ForwardRef(Button))
              color="primary"
              data-testid="submit-community"
              fullWidth={true}
              onClick={[MockFunction]}
              variant="contained"
            >
              Add community
            </WithStyles(ForwardRef(Button))>
          </WithStyles(ForwardRef(Grid))>
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Grid))>
    </form>
  </ConfirmDiscardFormChanges>
</StandardPage>
`;

exports[`Community form should render form 1`] = `
<StandardPage
  title="Add a missing community"
>
  <ConfirmDiscardFormChanges
    dirty={true}
    submitSucceeded={false}
  >
    <form>
      <NavigationDialogBox
        when={true}
      />
      <WithStyles(ForwardRef(Grid))
        container={true}
        spacing={3}
      >
        <WithStyles(ForwardRef(Grid))
          item={true}
          xs={12}
        >
          <WithStyles(ForwardRef(Grid))
            item={true}
            xs={12}
          >
            <StandardCard
              help={Object {}}
              title="Community details"
            >
              <WithStyles(ForwardRef(Grid))
                container={true}
                spacing={3}
              >
                <WithStyles(ForwardRef(Grid))
                  item={true}
                  xs={12}
                >
                  <Field
                    autoFocus={true}
                    component={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "propTypes": Object {
                          "classes": [Function],
                          "help": [Function],
                          "innerRef": [Function],
                          "textFieldId": [Function],
                        },
                        "render": [Function],
                      }
                    }
                    disabled={false}
                    fullWidth={true}
                    label="Title of community"
                    name="rek_title"
                    placeholder="Type a title for this community"
                    required={true}
                    textFieldId="rek-title"
                    type="text"
                    validate={
                      Array [
                        [Function],
                      ]
                    }
                  />
                </WithStyles(ForwardRef(Grid))>
                <WithStyles(ForwardRef(Grid))
                  item={true}
                  xs={12}
                >
                  <Field
                    component={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "propTypes": Object {
                          "classes": [Function],
                          "help": [Function],
                          "innerRef": [Function],
                          "textFieldId": [Function],
                        },
                        "render": [Function],
                      }
                    }
                    disabled={false}
                    fullWidth={true}
                    label="Community description"
                    multiline={true}
                    name="rek_description"
                    placeholder="Type a description of this community."
                    required={true}
                    rows={5}
                    textFieldId="rek-description"
                    validate={
                      Array [
                        [Function],
                      ]
                    }
                  />
                </WithStyles(ForwardRef(Grid))>
                <WithStyles(ForwardRef(Grid))
                  item={true}
                  xs={12}
                >
                  <WithStyles(ForwardRef(Typography))>
                    Add up to 10 individual keywords, or a pipe separated list, that describe the content of the thesis. (eg. one|two|three)
                  </WithStyles(ForwardRef(Typography))>
                  <Field
                    ListEditorForm={
                      Object {
                        "$$typeof": Symbol(react.memo),
                        "compare": null,
                        "type": [Function],
                      }
                    }
                    component={
                      Object {
                        "$$typeof": Symbol(react.memo),
                        "compare": null,
                        "type": [Function],
                      }
                    }
                    disabled={false}
                    listEditorId="rek-keywords"
                    locale={
                      Object {
                        "form": Object {
                          "locale": Object {
                            "addButtonLabel": "Add",
                            "id": "keywords-input",
                            "inputFieldHint": "Type keywords",
                            "inputFieldLabel": "Keywords",
                          },
                        },
                        "header": Object {
                          "locale": Object {
                            "deleteAll": "Remove all keywords",
                            "deleteAllConfirmation": Object {
                              "cancelButtonLabel": "No",
                              "confirmButtonLabel": "Yes",
                              "confirmationMessage": "Are you sure you want to delete all keywords?",
                              "confirmationTitle": "Delete all",
                            },
                            "nameColumn": "Keyword",
                            "reorderColumn": "Reorder keywords",
                          },
                        },
                        "row": Object {
                          "locale": Object {
                            "deleteHint": "Remove this keyword",
                            "deleteRecordConfirmation": Object {
                              "cancelButtonLabel": "No",
                              "confirmButtonLabel": "Yes",
                              "confirmationMessage": "Are you sure you want to delete this keyword?",
                              "confirmationTitle": "Delete keyword",
                            },
                            "moveDownHint": "Move keyword down the order",
                            "moveUpHint": "Move keyword up the order",
                          },
                        },
                      }
                    }
                    maxCount={10}
                    name="fez_record_search_key_keywords"
                    searchKey={
                      Object {
                        "order": "rek_keywords_order",
                        "value": "rek_keywords",
                      }
                    }
                  />
                </WithStyles(ForwardRef(Grid))>
                <WithStyles(ForwardRef(Grid))
                  item={true}
                  xs={12}
                >
                  <WithStyles(ForwardRef(Typography))>
                    Internal Notes (admin)
                  </WithStyles(ForwardRef(Typography))>
                  <Field
                    component={[Function]}
                    disabled={false}
                    fullWidth={true}
                    label="Internal Notes (admin)"
                    multiline={true}
                    name="internalNotes"
                    placeholder="Optionally enter internal notes"
                    richEditorId="internalNotes"
                    rows={5}
                  />
                </WithStyles(ForwardRef(Grid))>
              </WithStyles(ForwardRef(Grid))>
            </StandardCard>
          </WithStyles(ForwardRef(Grid))>
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          container={true}
          spacing={2}
        >
          <WithStyles(ForwardRef(Grid))
            item={true}
            sm={true}
            xs={false}
          />
          <WithStyles(ForwardRef(Grid))
            item={true}
            sm="auto"
            xs={12}
          >
            <WithStyles(ForwardRef(Button))
              data-testid="cancel-community"
              disabled={false}
              fullWidth={true}
              onClick={[Function]}
              variant="contained"
            >
              Cancel and return to the homepage
            </WithStyles(ForwardRef(Button))>
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid))
            item={true}
            sm="auto"
            xs={12}
          >
            <WithStyles(ForwardRef(Button))
              color="primary"
              data-testid="submit-community"
              fullWidth={true}
              onClick={[MockFunction]}
              variant="contained"
            >
              Add community
            </WithStyles(ForwardRef(Button))>
          </WithStyles(ForwardRef(Grid))>
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Grid))>
    </form>
  </ConfirmDiscardFormChanges>
</StandardPage>
`;

exports[`Community form should render success panel 1`] = `
<StandardPage
  title="Add a missing community"
>
  <WithStyles(ForwardRef(Grid))
    container={true}
    spacing={3}
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={12}
    >
      <StandardCard
        title="Community added successfully"
      >
        <WithStyles(ForwardRef(Typography))>
          Your new community was created successfully.
        </WithStyles(ForwardRef(Typography))>
      </StandardCard>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    container={true}
    spacing={2}
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={true}
    />
    <WithStyles(ForwardRef(Grid))
      item={true}
    >
      <WithStyles(ForwardRef(Button))
        fullWidth={true}
        onClick={[Function]}
        variant="contained"
      >
        Return to the homepage
      </WithStyles(ForwardRef(Button))>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
    >
      <WithStyles(ForwardRef(Button))
        color="primary"
        fullWidth={true}
        onClick={[Function]}
        variant="contained"
      >
        Add another community
      </WithStyles(ForwardRef(Button))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</StandardPage>
`;
